### PR TITLE
Implemented Issue #39 & altered Author Model to use url as primary key  instead of UUID 

### PR DESCRIPTION
**Pull request by XuechunHou**

### CODE REVIEW

**Comment by XuechunHou** [comment]
Also fixed service/<str:author1_id>/friends/<path:author2_id>/ endpoint so that author2_id now is a url field.

File: web_server/authors/views.py [hunk 1]
```diff
@@ -202,25 +206,29 @@ def post_creation_and_retrival_to_curr_auth_user(request):
...
-        new_post.content     = post['content']      #: "stuffs",
-        new_post.author      = request.user         # the authenticated user
+        new_post.content = post['content']      #: "stuffs",
+        new_post.author = request.user         # the authenticated user
         # Categories added after new post is saved
-        new_post.count       = 0                    #: 1023, initially the number of comments is zero
-        new_post.size        = size                 #: 50,
+        #: 1023, initially the number of comments is zero
+        new_post.count = 0
+        new_post.size = size                 #: 50,
```
Comment (AustinGrey) [inline web_server/authors/views.py:243]: This looks like you've been using automatic code style fixing tools on the entire file whenever you make changes. Would you mind only doing this for code you change in the future? The style you have isn't always the most readable style and adds lots of changes to your commits that make it hard to tell what is going on.

File: web_server/authors/views.py [hunk 2]
```diff
@@ -8,45 +8,41 @@
...
+# used for stripping url protocol
+url_regex = re.compile(r"(http(s?))?://")
 # Ida Hou
 # return a list of author id that are currently stored in database and
 # are not friend with current author
+
+
 def view_list_of_available_authors_to_befriend(request, author_id):
     if request.method != 'GET':
         return HttpResponse("Method not Allowed", status=405)
```
Comment (AustinGrey) [inline web_server/authors/views.py]: Did you see the Endpoint and Handler classes? They abstract the simple error handling like this away so your handlers become more compact. Is there a reason you didn't or don't want to use them?

**Review by AustinGrey — COMMENTED** [review]
Just the two comments I had to give, please let me know your thoughts. Also, I wouldn't merge this until PR #104 gets approved, otherwise your changes will not deploy to heroku.

File: web_server/friendship/tests.py [hunk 3]
```diff
@@ -1,3 +1,77 @@
-from django.test import TestCase
+from django.test import TestCase, Client
```
Comment (AustinGrey) [inline web_server/friendship/tests.py]: I thought we had discussed using the unittest library for tests. Are you expecting us to use django tests for tests?

**Review by AustinGrey — COMMENTED** [review]
You currently are experiencing merge conflicts due to PR's that were submitted before yours. Please fix the merge conflicts.

File: web_server/friendship/tests.py (hunk 3 above)
Comment (AustinGrey) [reply web_server/friendship/tests.py]: Nevermind, I see now in @hyeonyu1 PR #109 that we are using django tests.

**Review by AustinGrey — APPROVED** [review]
I've looked it over and everything looks code from a code perspective. Please feel free to merge if this is working fine on your local machine.

**Pull request merged**